In [1]:
import pandas as pd

In [2]:
data=pd.read_csv('C:/Users/ronit/Downloads/employees.csv')

In [3]:
data

First Name  Gender  Start Date Last Login Time  Salary  Bonus %  \
0      Douglas    Male    8/6/1993        12:42 PM   97308    6.945   
1       Thomas    Male   3/31/1996         6:53 AM   61933    4.170   
2        Maria  Female   4/23/1993        11:17 AM  130590   11.858   
3        Jerry    Male    3/4/2005         1:00 PM  138705    9.340   
4        Larry    Male   1/24/1998         4:47 PM  101004    1.389   
..         ...     ...         ...             ...     ...      ...   
995      Henry     NaN  11/23/2014         6:09 AM  132483   16.655   
996    Phillip    Male   1/31/1984         6:30 AM   42392   19.675   
997    Russell    Male   5/20/2013        12:39 PM   96914    1.421   
998      Larry    Male   4/20/2013         4:45 PM   60500   11.985   
999     Albert    Male   5/15/2012         6:24 PM  129949   10.169   

    Senior Management                  Team  
0                True             Marketing  
1                True                   NaN  
2               False               Finance  
3                True               Finance  
4                True       Client Services  
..                ...                   ...  
995             False          Distribution  
996             False               Finance  
997             False               Product  
998             False  Business Development  
999              True                 Sales  

[1000 rows x 8 columns]

In [15]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   First Name         933 non-null    object 
 1   Gender             855 non-null    object 
 2   Start Date         1000 non-null   object 
 3   Last Login Time    1000 non-null   object 
 4   Salary             1000 non-null   int64  
 5   Bonus %            1000 non-null   float64
 6   Senior Management  933 non-null    object 
 7   Team               957 non-null    object 
dtypes: float64(1), int64(1), object(6)
memory usage: 62.6+ KB


In [16]:
data.isnull().sum()

First Name            67
Gender               145
Start Date             0
Last Login Time        0
Salary                 0
Bonus %                0
Senior Management     67
Team                  43
dtype: int64

in our given data set the first coloumn ha 67 missing values. second coloumn that is gender has 145 missing values and so on. missing values is a problem for annalising data. It gives wrong results. so we have to eleminate the missing values.

In [29]:
update_data=data.dropna()
update_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 764 entries, 0 to 999
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   First Name         764 non-null    object 
 1   Gender             764 non-null    object 
 2   Start Date         764 non-null    object 
 3   Last Login Time    764 non-null    object 
 4   Salary             764 non-null    int64  
 5   Bonus %            764 non-null    float64
 6   Senior Management  764 non-null    object 
 7   Team               764 non-null    object 
dtypes: float64(1), int64(1), object(6)
memory usage: 53.7+ KB


In [30]:
update_data.isnull().sum()

First Name           0
Gender               0
Start Date           0
Last Login Time      0
Salary               0
Bonus %              0
Senior Management    0
Team                 0
dtype: int64

In [31]:
update_data.describe()

Salary     Bonus %
count     764.000000  764.000000
mean    90433.196335   10.148041
std     32864.665282    5.608733
min     35013.000000    1.015000
25%     62071.750000    5.193250
50%     90428.000000    9.658500
75%    118075.250000   14.965000
max    149908.000000   19.944000

In [32]:
update_data['Team'].describe()

count                      764
unique                      10
top       Business Development
freq                        88
Name: Team, dtype: object

In [33]:
dummy=pd.get_dummies(update_data['Team'])
dummy

Business Development  Client Services  Distribution  Engineering  \
0                       0                0             0            0   
2                       0                0             0            0   
3                       0                0             0            0   
4                       0                1             0            0   
5                       0                0             0            0   
..                    ...              ...           ...          ...   
994                     0                0             0            0   
996                     0                0             0            0   
997                     0                0             0            0   
998                     1                0             0            0   
999                     0                0             0            0   

     Finance  Human Resources  Legal  Marketing  Product  Sales  
0          0                0      0          1        0      0  
2          1                0      0          0        0      0  
3          1                0      0          0        0      0  
4          0                0      0          0        0      0  
5          0                0      1          0        0      0  
..       ...              ...    ...        ...      ...    ...  
994        0                0      0          1        0      0  
996        1                0      0          0        0      0  
997        0                0      0          0        1      0  
998        0                0      0          0        0      0  
999        0                0      0          0        0      1  

[764 rows x 10 columns]

In [21]:
dummy.isnull().sum()

Female    0
Male      0
dtype: int64

In [23]:
dummy.iloc[995]

Female    0
Male      0
Name: 995, dtype: uint8

In [ ]:
updated